python/shotgun_data/shotgun_data_retriever.py

```diff
@@ -891,17 +891,26 @@ def _get_thumbnail_path(url, bundle, directory_only=False):
             #   /bundle_cache_location/thumbs/C1C2/C3C4/rest_of_hash.*
             cache_matches = glob.glob("%s.*" % cache_base)
             if len(cache_matches):
+                if len(cache_matches) > 1:
```

Comment:
If i am reading this correctly, it seems like we are removing a `*.jpeg` file only to replace it with exactly the same `*.jpg` file? So this change in the shotgunutils framework will basically mean that all thumbnail cache are invalid and will be re-downloaded again. This doesn't seem great to me - why cannot this change be backwards compatible?

I thought we wanted to add the ability to download and cache `png` and other formats alongside the existing `jpeg` support. Unless we have strong reasons not to (and in that case let's discuss) the behaviour for caching `jpeg` files should remain unchanged.